### PR TITLE
add optional user credential options

### PR DESCRIPTION
**Pull request by ekmobile**

### CODE REVIEW

**Comment by mkloubertego** [comment]
@ekmobile 

following questions:

- can you also update `typedoc` in [package.json](https://github.com/egomobile/node-nats/blob/08dd6cd5f9cb902855bd5704348e95701f1b035d/package.json#L59) because there is an `npm audit` issue?
- does it make sense to add one or more options, especially the `tls` of `nats.StanOptions`?

**Comment by mkloubertego** [comment]
@ekmobile 

also describe the new environment variables in the following example code of the [README.md](https://github.com/egomobile/node-nats/blob/master/README.md#usage)

**Comment by mkloubertego** [comment]
@ekmobile 

and also describe environment variables here: https://github.com/egomobile/node-nats/blob/08dd6cd5f9cb902855bd5704348e95701f1b035d/src/classes/NatsClient.ts#L46

File: src/classes/NatsClient.ts [hunk 1]
```diff
@@ -169,9 +181,19 @@ export class NatsClient {
                     throw new Error('No serverURL defined');
                 }
 
-                const newClient = nats.connect(clusterId, clientId, {
+                let opts: nats.StanOptions = {
```
[the remark this answers is not in the document]
Comment (ekmobile) [reply src/classes/NatsClient.ts:184]: How should I set the properties if the opts are `const`?

**Comment by mkloubertego** [comment]
@ekmobile 

can you

**Pull request merged**